community/kernel/src/main/java/org/neo4j/kernel/impl/store/PropertyType.java

```diff
@@ -159,6 +159,10 @@ else if ( itemType <= GEOMETRY.byteValue() )
             {
                 return headOf( recordBytes, DynamicArrayStore.GEOMETRY_HEADER_SIZE );
             }
+            else if ( itemType <= TEMPORAL.byteValue() )
```

Comment:
The `<=` comparison looks weird. I see that this is most likely a copy-paste issue stemming from the `DOUBLE` comparison above, which `GEOMETRY` also copy-pasted. The `DOUBLE` comparison is, I believe, a special case for covering all integral and floating point types in one comparison, knowing that the `DOUBLE` `byteValue()` is the largest of those. I think it makes sense to change this condition to `==` and potentially also that of `GEOMETRY` while you're at it.